sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveTranspose.scala

```diff
@@ -57,8 +58,8 @@ class ResolveTranspose(sparkSession: SparkSession) extends Rule[LogicalPlan] {
           throw new AnalysisException(
             errorClass = "TRANSPOSE_NO_LEAST_COMMON_TYPE",
             messageParameters = Map(
-              "dt1" -> dt1.sql,
-              "dt2" -> dt2.sql)
+              "dt1" -> toSQLType(dt1.sql),
+              "dt2" -> toSQLType(dt2.sql))
```

Comment:
`toSQLType` over strings is some kind of workaround. In general case, let's leave convertions to strings intirely to `toSQLType`:
```suggestion
              "dt1" -> toSQLType(dt1),
              "dt2" -> toSQLType(dt2))
```